File: benchmarks/ruby/list-operations.rb
def make_list(n)
  acc = []
  (1..n).each { |i| acc.unshift(i) }
  acc
end

def sum_list(lst)
  return 0 if lst.empty?
  lst.first + sum_list(lst[1..-1])
end

def reverse_list(lst)
  acc = []
  lst.each { |item| acc.unshift(item) }
  acc
end

def list_benchmark(n)
  lst = make_list(n)
  sum = sum_list(lst)
  rev = reverse_list(lst)
  rev.length
end

list_benchmark(1000)

File: benchmarks/python/list-operations.py
def make_list(n):
    acc = []
    for i in range(n, 0, -1):
        acc.insert(0, i)
    return acc

def sum_list(lst):
    total = 0
    for item in lst:
        total += item
    return total

def reverse_list(lst):
    acc = []
    for item in lst:
        acc.insert(0, item)
    return acc

def list_benchmark(n):
    lst = make_list(n)
    total = sum_list(lst)
    rev = reverse_list(lst)
    return len(rev)

list_benchmark(1000)